doc: Update and improve files.md

hebasto opened the pull request:
This PR adds to the `files.md`:
- the `signet` subdirectory
- the `ip_asn.map` file
- some small improvements

hebasto committed:
doc: Update and improve files.md

Added the `signet` subdirectory and the `ip_asn.map` file.

practicalswift commented:
ACK 00a3b45696fea7ad3a6c00b769ec6c2fc53c1b5c

Thanks for improving documentation!

jonatack commented on doc/files.md:
```diff
@@ -26,40 +26,42 @@ Linux    | `$HOME/.bitcoin/`
 Windows  | `%APPDATA%\Bitcoin\` <sup>[\[1\]](#note1)</sup>
 
-2. The non-default data directory path can be specified by `-datadir` option.
+2. The data directory path can be specified with `-datadir` option.
```
s/with/with the/

s/option/configuration option/

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/20076#issuecomment-703294876).

jonatack commented on doc/files.md:
```diff
@@ -26,40 +26,42 @@ Linux    | `$HOME/.bitcoin/`
+Chain option                   | Data directory path
+-------------------------------|------------------------------
+`-chain=main` (by default)     | *path_to_datadir*`/`
+`-chain=signet` or `-signet`   | *path_to_datadir*`/signet/`
```
generally, the chain order in the codebase is main/test/signet/regtest

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/20076#issuecomment-703294876).

jonatack commented on doc/files.md:
```diff
@@ -26,40 +26,42 @@ Linux    | `$HOME/.bitcoin/`
 Subdirectory       | File(s)               | Description
 -------------------|-----------------------|------------
-`blocks/`          |                       | Blocks directory; can be specified by `-blocksdir` option (except for `blocks/index/`)
+`blocks/`          |                       | Blocks directory. Path can be specified with `-blocksdir` option (except for `blocks/index/`)
```
s/. Path/; path/ (to avoid full sentence structure and need for EOL full stop)

s/with/by/ or /with/with the/

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/20076#issuecomment-703294876).

jonatack commented on doc/files.md:
```diff
@@ -26,40 +26,42 @@ Linux    | `$HOME/.bitcoin/`
 `blocks/`          | `blkNNNNN.dat`<sup>[\[2\]](#note2)</sup> | Actual Bitcoin blocks (in network format, dumped in raw on disk, 128 MiB per file)
 `blocks/`          | `revNNNNN.dat`<sup>[\[2\]](#note2)</sup> | Block undo data (custom format)
-`chainstate/`      | LevelDB database      | Blockchain state (a compact representation of all currently unspent transaction outputs and some metadata about the transactions they are from)
+`chainstate/`      | LevelDB database      | Blockchain state (a compact representation of all currently unspent transaction outputs (UTXO) and some metadata about the transactions they are from)
```
s/UTXO/UTXOs/ and omit parasitic "some"

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/20076#issuecomment-703294876).

jonatack commented on doc/files.md:
```diff
@@ -26,40 +26,42 @@ Linux    | `$HOME/.bitcoin/`
-`-regtest`          | *path_to_datadir*`/regtest/`
+Chain option                   | Data directory path
+-------------------------------|------------------------------
+`-chain=main` (by default)     | *path_to_datadir*`/`
```
can omit "by "

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/20076#issuecomment-703294876).

jonatack reviewed (commented):
Concept ACK. Not a fan of many of the grammar changes that are both a bit noisy and which partially convert to full sentence structure and grammar without adding the articles ("the") and full stops the conversion implies. A few examples below, but I stopped after realising that many of the changes involve the same issues. Would prefer just the essential content changes without the grammar ones.

hebasto commented:
Updated 00a3b45696fea7ad3a6c00b769ec6c2fc53c1b5c -> 756c9dcfa9e9b609ad7dd07c1e802c9f0a65c743 ([pr20076.01](https://github.com/hebasto/bitcoin/commits/pr20076.01) -> [pr20076.02](https://github.com/hebasto/bitcoin/commits/pr20076.02), [diff](https://github.com/hebasto/bitcoin/compare/pr20076.01..pr20076.02)):

- addressed @jonatack's comments, dropped unneeded changes

jonatack commented on doc/files.md:
```diff
@@ -26,15 +26,16 @@ Linux    | `$HOME/.bitcoin/`
 Windows  | `%APPDATA%\Bitcoin\` <sup>[\[1\]](#note1)</sup>
 
-2. The non-default data directory path can be specified by `-datadir` option.
+2. The data directory path can be specified by `-datadir` option.
```
Since this is a full sentence, I guess it should have s/by/by the/.

Maybe

"A custom data directory path can be specified by the `-datadir` option."

or

"A non-default data directory path can be specified by the `-datadir` option."

hebasto replied:
Thankc! [Updated](https://github.com/bitcoin/bitcoin/pull/20076#issuecomment-703297165).

jonatack reviewed (commented):
Oh much better, thanks @hebasto 

ACK mod one comment

hebasto commented:
Updated 756c9dcfa9e9b609ad7dd07c1e802c9f0a65c743 -> 8d1ab6c7330d79e4535c8fe0adbbc0df22c1e794 ([pr20076.02](https://github.com/hebasto/bitcoin/commits/pr20076.02) -> [pr20076.03](https://github.com/hebasto/bitcoin/commits/pr20076.03), [diff](https://github.com/hebasto/bitcoin/compare/pr20076.02..pr20076.03)):

- addressed @jonatack's [comment](https://github.com/bitcoin/bitcoin/pull/20076#discussion_r499274519)

jonatack commented:
ACK

practicalswift commented:
ACK 8d1ab6c7330d79e4535c8fe0adbbc0df22c1e794

promag reviewed (commented):
Concept ACK

hebasto commented:
@naumenkogs Mind looking into this doc update as it is about `ip_asn.map` file?

naumenkogs commented:
Everything looks good.

W.r.t asmap, maybe instead of linking a wikipedia article we should link some asmap-related info? I couldn't find a good webpage to link, so maybe just add "Experimental feature: see [asmap](https://github.com/bitcoin/bitcoin/blob/42065518d937e036cce6b0ad21890f8fc2c50cfc/doc/release-notes/release-notes-0.20.0.md#new-settings)"

hebasto commented:
Updated 8d1ab6c7330d79e4535c8fe0adbbc0df22c1e794 -> 2dc79c4264d608ebe48c980f0ead54274ab3ee4f ([pr20076.03](https://github.com/hebasto/bitcoin/commits/pr20076.03) -> [pr20076.04](https://github.com/hebasto/bitcoin/commits/pr20076.04), [diff](https://github.com/hebasto/bitcoin/compare/pr20076.03..pr20076.04)).

@naumenkogs 
> W.r.t asmap, maybe instead of (or in addition to) linking a wikipedia article we should link some asmap-related info? I couldn't find a good webpage to link, so maybe just add "Experimental feature: see [asmap-CLI](https://github.com/bitcoin/bitcoin/blob/42065518d937e036cce6b0ad21890f8fc2c50cfc/doc/release-notes/release-notes-0.20.0.md#new-settings) and [asmap-utility](https://github.com/bitcoin/bitcoin/pull/18573)"

Reworked. I don't think adding link to the feature description is the purpose of this file.

naumenkogs commented:
ACK

decryp2kanon commented:
Concept ACK

practicalswift commented:
ACK 2dc79c4264d608ebe48c980f0ead54274ab3ee4f

MarcoFalke commented:
ACK 2dc79c4264d608ebe48c980f0ead54274ab3ee4f
